modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/autocomplete/Autocomplete.js

```diff
@@ -12,346 +12,129 @@
 import {showErrorNotification} from '../../utilities/notifications';
 import InfiniteScroller from '../infinite_scroller/InfiniteScroller';
 
-function Autocomplete({onChange, onItemsUpdated, onValueUpdated, ...props}) {
-	const [query, setQuery] = useState(props.initialLabel || '');
-	const [initialised, setInitialised] = useState(
-		Boolean(props.customViewModuleUrl || props.customView)
-	);
-	const [active, setActive] = useState(false);
-	const [selectedItem, updateSelectedItem] = useState(props.initialValue);
-	const [items, updateItems] = useState(null);
+function Autocomplete({
+	apiUrl,
+	customView: CustomView,
+	disabled = false,
+	inputName,
+	inputPlaceholder = Liferay.Language.get('type-here'),
+	valueKey,
+	labelKey,
+	pageSize = 10,
+}) {
+	const [query, setQuery] = useState('');
+	const [selectedItem, setSelectedItem] = useState(null);
+	const [items, setItems] = useState(null);
 	const [loading, setLoading] = useState(false);
-	const [totalCount, updateTotalCount] = useState(null);
-	const [lastPage, updateLastPage] = useState(null);
-	const [page, updatePage] = useState(1);
-	const [pageSize, updatePageSize] = useState(props.pageSize);
-	const node = useRef();
-	const dropdownNode = useRef();
-	const inputNode = useRef();
-	const FetchedCustomView = useLiferayModule(props.customViewModuleUrl);
+	const [totalCount, setTotalCount] = useState(null);
+	const [lastPage, setLastPage] = useState(null);
+	const [page, setPage] = useState(1);
+	const [internalPageSize, setInternalPageSize] = useState(pageSize);
 	const isMounted = useIsMounted();
-	const [debouncedGetItems, updateDebouncedGetItems] = useState(null);
-
-	useEffect(() => {
-		updateDebouncedGetItems(() =>
-			debouncePromise(getData, props.fetchDataDebounce)
-		);
-	}, [props.fetchDataDebounce]);
-
-	const currentValue = selectedItem
-		? getValueFromItem(selectedItem, props.itemsKey)
-		: null;
-	const currentLabel = selectedItem
-		? getValueFromItem(selectedItem, props.itemsLabel)
-		: null;
-
-	const CustomView = props.customView || FetchedCustomView;
-
-	useEffect(() => {
-		if (items && items.length === 1 && props.autofill) {
-			const firstItem = items[0];
-			updateSelectedItem(firstItem);
-		}
-	}, [items, props.autofill, props.itemsKey, props.itemsLabel]);
-
-	useEffect(() => {
-		const value =
-			selectedItem && getValueFromItem(selectedItem, props.itemsKey);
-
-		if (props.id) {
-			Liferay.fire(AUTOCOMPLETE_VALUE_UPDATED, {
-				id: props.id,
-				itemData: selectedItem,
-				value,
-			});
-		}
-
-		if (onValueUpdated) {
-			onValueUpdated(value, selectedItem);
-		}
 
-		if (onChange) {
-			onChange({target: {value}});
-		}
-	}, [selectedItem, props.id, props.itemsKey, onChange, onValueUpdated]);
-
-	useEffect(() => {
-		if (query) {
-			setInitialised(true);
-		}
-
-		if (props.infiniteScrollMode) {
-			updateItems(null);
-		}
-
-		updatePage(1);
-		updateTotalCount(null);
-		updateLastPage(null);
-	}, [props.infiniteScrollMode, query]);
```

Comment:
`infiniteScrollMode ` is never toggled and I moved some of this logic to where query is changed. It is better to update state in a central location rather than updating it once and then updating it again on state change.

1 render is better than 2 renders